### ISSUE TITLE
Incorrect marker positions in contained map

### ISSUE THREAD

**Issue by indapublic** [google-map-react/google-map-react#523]
`"google-map-react": "^0.32.0"`

Full screen mode (sorry, markers are small):
<img width="960" alt="screen shot 2018-03-07 at 22 07 50" src="https://user-images.githubusercontent.com/856260/37091540-43f8c91c-2254-11e8-8438-10d5f77411c7.png">

Contained mode:
![screen shot 2018-03-07 at 22 08 01](https://user-images.githubusercontent.com/856260/37091551-5405b2b6-2254-11e8-9d33-324e2e3b529e.png)

I found problem #464 with fullscreen mode (which has been fixed), probably it can be related to it?

**Comment by itsmichaeldiego** [comment]
@indapublic This is not reproducible for me in my map, could you share your code and the css styles applied to the map?

**Comment by indapublic** [comment]
Source of component who render map with current geolocation:

```
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import GoogleMapReact from 'google-map-react'
import { geolocated } from 'react-geolocated'

import Markers from './Markers/'

import { google } from 'config.env'

class Map extends Component {
	static propTypes = {
		center: PropTypes.object
	}

	render() {
		const { children, isGeolocationAvailable, coords, center } = this.props
		return (
			<div className="Map">
				<GoogleMapReact
					bootstrapURLKeys={{
						key: [google.map.apiKey]
					}}
					center={
						!!center
							? center
							: !!isGeolocationAvailable && !!coords
								? {
										lat: coords.latitude,
										lng: coords.longitude
								  }
								: {
										lat: 59.95,
										lng: 30.33
								  }
					}
					defaultZoom={11}
					options={{
						fullscreenControl: false,
						minZoom: 4,
						maxZoom: 20
					}}
				>
					{children}
					{!!isGeolocationAvailable &&
						!!coords && (
							<Markers.Me
								lat={coords.latitude}
								lng={coords.longitude}
								text="Я"
							/>
						)}
				</GoogleMapReact>
			</div>
		)
	}
}

export default geolocated({
	positionOptions: {
		enableHighAccuracy: true
	},
	userDecisionTimeout: 5000
})(Map)

```

**Comment by indapublic** [comment]
Map markers:

<img
  lat={lat}
  lng={lng}
  onClick={onClick}
  className="MapMarkerImage"
  src={addressImage}
  alt={title}
/>

**Comment by indapublic** [comment]
Map style

```
.Map {
    width: 100%;
    min-height: 400px;
    height: 600px;
}
```

**Comment by indapublic** [comment]
I noticed that it does not always appear. Sometimes position is fixed as it should, either immediately or after playing with the zoom or map moving

**Comment by itsmichaeldiego** [comment]
@indapublic I can't reproduce this really, I can place the map anywhere and it works perfectly fine, if you can create a codepen or something similar live code would be great!

**Comment by indapublic** [comment]
@itsmichaeldiego Ok, I'll try to make example sometime later

**Comment by itsmichaeldiego** [comment]
@indapublic That would be great, thanks. My map is working fine either way so I cannot really help you this time.

**Comment by indapublic** [comment]
I found a situation when the positions are wrong. But will create example on weekend. But try to describe now:
We have container (as flex) with map and something more (some popup in my case)
If the popup is invisible, the map occupies the entire space - all works

![screen shot 2018-03-16 at 08 35 14](https://user-images.githubusercontent.com/856260/37494549-04e21510-28f5-11e8-8f38-02daed11051c.png)

If the window is open, the card occupies part of the container - All marker positions will be biased towards for popup width

![screen shot 2018-03-16 at 08 36 39](https://user-images.githubusercontent.com/856260/37494719-d7687e20-28f5-11e8-8f6c-475965f57279.png)

**Comment by indapublic** [comment]
Map container with `padding: 0`

![screen shot 2018-03-16 at 09 14 47](https://user-images.githubusercontent.com/856260/37495783-b77b961a-28fa-11e8-8f67-0a234c9bfcde.png)

Map container with `padding: 200px`

<img width="841" alt="screen shot 2018-03-16 at 09 16 54" src="https://user-images.githubusercontent.com/856260/37495796-cf713338-28fa-11e8-8dc0-ad3d86a96312.png">

I'm not sure that behaviour is expected

**Comment by itsmichaeldiego** [comment]
@indapublic Oh, yeah, I see what's going on in here. The problem is that you're applying padding to the container of the map, did you try changing its `width` instead? I think there is a problem on the way you're handling this card via css.

**Comment by itsmichaeldiego** [comment]
@indapublic Actually, did you try with `resetBoundsOnResize?`

https://github.com/istarkov/google-map-react/blob/master/DOC.md#simple-example

> By default map will not raise onChange event if parent size has changed, to change such behavior add resetBoundsOnResize = {true} property.

**Comment by indapublic** [comment]
@itsmichaeldiego I will check `resetBoundsOnResize`, thanks.

As I said before `padding` is not only issue. [Try to place map with any div inside flexbox](https://github.com/istarkov/google-map-react/issues/523#issuecomment-373546009) and you see the difference.

I fixed it issue with layout changing at the moment btw

**Comment by itsmichaeldiego** [comment]
@indapublic Please let me know if it works so I can close the issue, thanks in advance.

**Comment by marharyta** [comment]
I am having a similar issue with my map. When I zoom out the markers take weird position below the actual coordinates. Please, check the demo [https://codesandbox.io/s/jjw9pmvpz5](https://codesandbox.io/s/jjw9pmvpz5) here. @itsmichaeldiego

**Comment by itsmichaeldiego** [comment]
@marharyta The first codesandbox you provided does not work, but I think your problem is the position of the markers, by default they're aligned top-left, so if your markers are mean to be aligned in the center, you might need to add styles to position it:

https://github.com/google-map-react/google-map-react/blob/master/API.md#positioning-a-marker

**Comment by itsmichaeldiego** [comment]
@marharyta Yeah, it is directly related to the marker position, our markers are by default positioned at top-left, here is the fix: https://codesandbox.io/s/n9lrxw0670

**Comment by marharyta** [comment]
@itsmichaeldiego thanks a lot! I am not sure about default, since a lot of people will be pointing to the locations in different ways, yet i believe it doesn't hurt to put more emphasis on that in the documentation.

**Comment by marharyta** [comment]
@itsmichaeldiego And yes, it helped and it works! Thanks a lot!

**Comment by itsmichaeldiego** [comment]
@marharyta Ole hyvä! You are more than welcome! I agree, it would also be cold to be able to set the position by parameter, something like:

```javascript
<Marker
  lat={latitude}
  lng={longitude}
  position="top-right"
/>
```

**Comment by itsmichaeldiego** [comment]
@indapublic This should be totally fixed in `1.0.5` as we're calculating the marker's position differently, let me know if otherwise and we can re-open and re-visit the issue.
